### PR TITLE
Feature/multi component setup

### PR DESCRIPTION
**Pull request by Robert-SD**

### CODE REVIEW

File: android/src/main/kotlin/com/adyen/checkout/flutter/PlatformApi.kt
Comment (Robert-SD) [inline android/src/main/kotlin/com/adyen/checkout/flutter/PlatformApi.kt:1]: Generated code - no review required.

File: ios/Classes/PlatformApi.swift
Comment (Robert-SD) [inline ios/Classes/PlatformApi.swift:1]: Generated code - no review required.

File: lib/src/generated/platform_api.g.dart
Comment (Robert-SD) [inline lib/src/generated/platform_api.g.dart:1]: Generated code - no review required.

File: android/src/main/kotlin/com/adyen/checkout/flutter/components/googlepay/GooglePayComponentManager.kt [hunk 1]
```diff
@@ -13,13 +16,22 @@ import com.adyen.checkout.flutter.utils.ConfigurationMapper.mapToGooglePayConfig
...
 
 class GooglePayComponentManager(
     private val activity: FragmentActivity,
     private val sessionHolder: SessionHolder,
     private val componentFlutterInterface: ComponentFlutterInterface,
 ) {
-    private val googlePayComponents: MutableList<BaseGooglePayComponent> = mutableListOf()
+    private var googlePayComponent: BaseGooglePayComponent? = null
+
+    companion object {
```
Comment (Robert-SD) [inline android/src/main/kotlin/com/adyen/checkout/flutter/components/googlepay/GooglePayComponentManager.kt:30]: Better to inject not static

Comment (Robert-SD) [reply android/src/main/kotlin/com/adyen/checkout/flutter/components/googlepay/GooglePayComponentManager.kt:30]: Remove

File: android/src/main/kotlin/com/adyen/checkout/flutter/components/googlepay/GooglePayComponentManager.kt [hunk 2]
```diff
@@ -103,25 +107,23 @@ class GooglePayComponentManager(
...
     }
 
-    fun onDispose(componentId: String) {
-        googlePayComponents.firstOrNull { it.componentId == componentId }.apply {
-            this?.dispose()
-            googlePayComponents.remove(this)
+    fun onDispose() {
+        googlePayComponent?.apply {
+            dispose()
+            googlePayComponent = null
```
Comment (araratthehero) [inline android/src/main/kotlin/com/adyen/checkout/flutter/components/googlepay/GooglePayComponentManager.kt:126]: I would write this
```suggestion
        googlePayComponent?.dispose()
        googlePayComponent = null            
```

**Review by araratthehero — DISMISSED** [review]

**Review by araratthehero — DISMISSED** [review]

**Review by araratthehero — DISMISSED** [review]

**Review by descorp — APPROVED** [review]

**Pull request merged**